source/includes/api/_query.md

```diff
@@ -1,204 +1,278 @@
-
-SELECT ST_valueCount(rast, 1, true) FROM table
+```shell
+curl -i -X GET 'http://api.resourcewatch.org/v1/query/098b33df-6871-4e53-a5ff-b56a7d989f9a?sql=SELECT cartodb_id, iso, name_0, name_1, type_1 FROM gadm28_adm1 limit 10'
 ```
 
-## From clause
+> The same query executed as a POST request providing the SQL as request body param:
 
-It is possible to refer to a dataset by using its **table name**, **slug** or **id**. 
-
-```sql
-SELECT * FROM <dataset.tableName>
-SELECT * FROM <dataset.slug>
-SELECT * FROM <dataset.id>
-```
-
-## Where clause
-
-```sql
-SELECT * FROM table WHERE a > 2
-SELECT * FROM table WHERE a = 2
-SELECT * FROM table WHERE a < 2
-SELECT * FROM table WHERE a >= 2
-SELECT * FROM table WHERE a = 2 and b < 2
-SELECT * FROM table WHERE text like 'a%'
-SELECT * FROM table WHERE st_intersects(st_setsrid(st_geomfromgeojson('{}'), 4326), the_geom)
-SELECT * FROM table WHERE data BETWEEN 1 AND 3
+```shell
+curl -i -X POST 'http://api.resourcewatch.org/v1/query/098b33df-6871-4e53-a5ff-b56a7d989f9a/' \
+-H 'Content-Type: application/json' \
+-d '{
+    "sql": "SELECT cartodb_id, iso, name_0, name_1, type_1 FROM gadm28_adm1 limit 10"
+}'
 ```
 
-## Group by clause
-
-It is possible to group results by **column name**, **SQL functions** and **PostGIS functions**.
-Special grouping operations are available for document-type datasets (CSV and JSON) - see [this link](https://github.com/NLPchina/elasticsearch-sql/tree/5.5.2.0#beyond-sql) for more info.
-
-
-```sql
-SELECT * FROM tablename GROUP BY columnOne
-SELECT * FROM tablename GROUP BY columnOne, columnTwo
-SELECT * FROM tablename GROUP BY ST_GeoHash(the_geom_point,8)
-
-SELECT a, count(int) FROM table GROUP BY a
-SELECT count(*) FROM tablename GROUP BY ST_GeoHash(the_geom, 8)
+> Example response:
+
+```json
+{
+    "data": [
+        {
+            "cartodb_id": 1830,
+            "iso": "MEX",
+            "name_0": "Mexico",
+            "name_1": "Ciudad de México",
+            "type_1": "Distrito Federal"
+        },
+        {
+            "cartodb_id": 1109,
+            "iso": "HTI",
+            "name_0": "Haiti",
+            "name_1": "L'Artibonite",
+            "type_1": "Département"
+        },
+        ...
+    ],
+    "meta": {
+        "cloneUrl": {
+            "http_method": "POST",
+            "url": "/dataset/098b33df-6871-4e53-a5ff-b56a7d989f9a/clone",
+            "body": {
+                "dataset": {
+                    "datasetUrl": "/query/098b33df-6871-4e53-a5ff-b56a7d989f9a?sql=SELECT%20cartodb_id%2C%20iso%2C%20name_0%2C%20name_1%2C%20type_1%20FROM%20gadm28_adm1%20limit%2010",
+                    "application": [
+                        "your",
+                        "apps"
+                    ]
+                }
+            }
+        }
+    }
+}
 ```
 
+To execute a query over a dataset's data, you can either perform a GET request providing the SQL query as query param, or a POST request providing the SQL query in the request body.
```

Comment:
Need to explicitly call out that the dataset id is part of the path.

Reply:
Overall, I think a little more description of how the dataset id shows up in the query is needed. Does it depend on the provider? If not the dataset id, what do I put in the FROM clause?

Reply:
From what I was able to understand so far, I think some providers are prepared to receive `dataset.id` either as query param (as in `/query/:datasetId`) or as part of the FROM clause (as in `FROM dataset.id`) - this is the case of document-based datasets.

However, other providers (such as Carto) always expect to receive dataset id as a query param and the dataset tableName in the FROM clause - and they are quite inflexible when it comes to this, i.e. any other format results in an error response.

So I'd suggest that we take `/query/:datasetId` + the FROM clause referring to the `dataset.tableName` as our officially supported way of querying datasets - and update the docs to match these expectations.

Does this sound good?

Also, @tiagojsag do you agree with this?

Reply:
hey,

yeah, having different approaches for different dataset types is a pain. However, this is live and out there, and people are using it, so we should do our best to document what we have. I understand and agree with the temptation to hide outliers, but since this is actually used, it may be confusing for people not to see what they are using covered in the docs

Reply:
Ok, I see your point - then I'll update the docs to take into account the different approaches for querying datasets and I'll include the known caveats I've found so far!

Reply:
Is it clearer this way (:point_down:)?
![Screenshot from 2020-05-12 12-36-44](https://user-images.githubusercontent.com/2981676/81682921-463cbe00-944d-11ea-8bf4-9c5f75ae694a.png)

Reply:
So just to confirm, `tableName` is sometimes the only options, but always acceptable in the `FROM` clause? Or are there providers where you can't use `tableName`?

Reply:
_(Answering your question, yes, `tableName` is always acceptable in the FROM clause, as long as the dataset id is provided in the path of the request)._

There was some confusion with this on my side, sorry for that. 

I changed a bit the way the docs are structured - now, initially, we have a "softer" introduction, suggesting the usage of `query/:datasetId` with the dataset `tableName` on the FROM clause, which works for all providers.

Then a bit below, I added a new section with alternative methods for querying, where I detail the different ways (and which providers support or not) these different approaches.

I hope these changes make this clearer :)